check mentions of formulas, specifiy backward compatibility

luisDVA opened the pull request:
addresses #1205 

Examples, vignettes, and readme no longer have mentions to formulas except for a more consistent message explaining that these may be used but only for backward compatibility with older versions of R.
* also fixed a typo or two

part of Tidyverse Dev Day 2025, ATL

jennybc commented:
I think it might be good to replace "older versions of R" with something specific, e.g. "R <= 4.1".

hadley commented:
I think we could probably just stick with the previous language and say "No longer recommended." (Sorry @luisDVA I think I gave you bad advice when we chatted)

hadley commented on R/detect.R:
```diff
@@ -6,8 +6,8 @@
-#'     argument. No longer recommended.
+#'   A formula, e.g. `~ .x + 1`. Only recommended if you require backward
+#'     compatibility with older versions of R.
 #'   * A string, integer, or list, e.g. `"idx"`, `1`, or `list("idx", 1)` which
```
There's also a mention of "lambda-formula" in the examples that should be removed.

luisDVA commented:
continuation of the work from dev day.
I reverted the comments about backward compatibility to just state "No longer recommended". The only mention of backward compatibility with R < 4.1 is now in a note at the end of the readme, as suggested by @hadley at the end of dev day. Also deleted a remaining mention of "lambda-formulas"
